library/Ivoz/Provider/Application/Service/User/SyncFromCsv.php

```diff
@@ -0,0 +1,119 @@
+<?php declare(strict_types=1);
+
+namespace Ivoz\Provider\Application\Service\User;
+
+use Ivoz\Core\Application\Service\EntityTools;
+use Ivoz\Provider\Domain\Model\Company\CompanyRepository;
+use Ivoz\Provider\Domain\Service\Ddi\DdiFactory;
+use Ivoz\Provider\Domain\Service\Extension\ExtensionFactory;
+use Ivoz\Provider\Domain\Service\Terminal\TerminalFactory;
+use Ivoz\Provider\Domain\Service\User\UserFactory;
+
+class SyncFromCsv
+{
+    private $companyRepository;
+    private $userFactory;
+    private $terminalFactory;
+    private $extensionFactory;
+    private $ddiFactory;
+    private $entityTools;
+
+    public function __construct(
+        CompanyRepository $companyRepository,
+        UserFactory $userFactory,
+        TerminalFactory $terminalFactory,
+        ExtensionFactory $extensionFactory,
+        DdiFactory $ddiFactory,
+        EntityTools $entityTools
+    ) {
+        $this->companyRepository = $companyRepository;
+        $this->userFactory = $userFactory;
+        $this->terminalFactory = $terminalFactory;
+        $this->extensionFactory = $extensionFactory;
+        $this->ddiFactory = $ddiFactory;
+        $this->entityTools = $entityTools;
+    }
+
+    /**
+     * @throws \Exception
+     *
+     * @return void
+     */
+    public function execute(int $companyId, string $csv)
+    {
+        $company = $this->companyRepository->find(
+            $companyId
+        );
+
+        if (!$company) {
+            throw new \DomainException(
+                'Company not found',
+                404
+            );
+        }
+
+        $rows = explode(
+            PHP_EOL,
+            trim($csv)
+        );
+
+        $errors = [];
+        foreach ($rows as $k => $row) {
+            $fields = str_getcsv(trim($row));
+            if (count($fields) < 11) {
+                $errors[$k+1] = 'Missing columns';
+                continue;
+            }
+
+            $user = array_splice($fields, 0, 3);
+            $terminal = array_splice($fields, 0, 4);
+            $extension = array_shift($fields);
+            $outboundDdi = $fields;
+
+            try {
+                $user = $this->userFactory->fromMassProvisioningCsv(
+                    $companyId,
+                    ...$user
+                );
```

Comment:
I think is better to use different variable names for parameters and output when they have different types (UserInterface vs Array in this case) 

What about userData for array sliced data? (Same applies to other arrays)